mission_control/navigator_missions/navigator_singleton/navigator.py

```diff
@@ -221,6 +238,57 @@ def get_response(self, **kwargs):
 
         return self.client(s_req)
 
+class MissionParam(object):
+    def __init__(self, nh, param, options):
+        self.nh = nh
+        self.param = param
+        self.options = options
+
+    @util.cancellableInlineCallbacks
+    def get(self):
+        # Returns deferred object, make sure to yield on this (same for below)
+        if not (yield self.exists()):
+          raise Exception("Mission Param {} not yet set".format(self.param))
+        value = yield self.nh.get_param(self.param)
+        if not self._valid(value):
+          raise Exception("Value {} is invalid for param {}. Valid values: {}".format(value, self.param, self.options))
+        else:
+          defer.returnValue(value)
+
+    @util.cancellableInlineCallbacks
+    def exists(self):
```

Comment:
You can just make this a normal function as follows (for cleanliness sake)
```python
# no decorator
def exists(self):
    return self.nh.has_param(self.param)
```
This will return a defer object that can be yeileded on. So calling the function is the same with less clutter in the function: `yield self.exists()`